File: Level1/신고_결과_받기/solution.go
import (
    "strings"
)

func solution(id_list []string, report []string, k int) []int {
    var ht map[string]int
    ht = make(map[string]int)
    id_list_len := len(id_list)
    flags := [][]bool {}
    counts := []int {}
    answer := []int {}
    for i, id := range id_list {
        ht[id] = i
        temp := []bool {}
        for j := 1; j <= id_list_len; j++ {
            temp = append(temp, false)
        }
        flags = append(flags, temp)
        counts = append(counts, 0)
        answer = append(answer, 0)
    }
    for _, rep := range report {
        rep := strings.Split(rep, " ")
        a := ht[rep[0]]
        b := ht[rep[1]]
        if (!flags[b][a]) {
            flags[b][a] = true
            counts[b]++
        }
    }
    for i, cnt := range counts {
        if (cnt >= k) {
            for j, f := range flags[i] {
                if (f) {
                    answer[j]++
                }
            }
        }
    }
    return answer
}

File: Level1/신고_결과_받기/solution.py
def solution(id_list, report, k):
    id_list_len = len(id_list)
    answer = [0] * id_list_len
    ht = dict(zip(id_list, range(id_list_len)))
    flags = [[False for _ in range(id_list_len)] for __ in range(id_list_len)]
    counts = [0] * id_list_len
    
    for rep in report:
        rep = [ht[r] for r in rep.split(" ")]
        if (flags[rep[1]][rep[0]] == False):
            flags[rep[1]][rep[0]] = True
            counts[rep[1]] += 1
    for i in range(id_list_len):
        if (counts[i] >= k):
            for j in range(id_list_len):
                if (flags[i][j] == True):
                    answer[j] += 1
    return answer